core/src/main/java/org/apache/iceberg/PartitionStatsUtil.java

```diff
@@ -0,0 +1,227 @@
+            7, Column.POSITION_DELETE_FILE_COUNT.name(), Types.IntegerType.get()),
+        Types.NestedField.optional(
+            8, Column.EQUALITY_DELETE_RECORD_COUNT.name(), Types.LongType.get()),
+        Types.NestedField.optional(
+            9, Column.EQUALITY_DELETE_FILE_COUNT.name(), Types.IntegerType.get()),
+        Types.NestedField.optional(10, Column.TOTAL_RECORD_COUNT.name(), Types.LongType.get()),
+        Types.NestedField.optional(11, Column.LAST_UPDATED_AT.name(), Types.LongType.get()),
+        Types.NestedField.optional(
+            12, Column.LAST_UPDATED_SNAPSHOT_ID.name(), Types.LongType.get()));
+  }
+
+  /**
+   * Creates an iterable of partition stats records from a given manifest file, using the specified
+   * table and record schema.
+   *
+   * @param table the table from which the manifest file is derived.
+   * @param manifest the manifest file containing metadata about the records.
+   * @param recordSchema the schema defining the structure of the records.
+   * @return a CloseableIterable of partition stats records as defined by the manifest file and
+   *     record schema.
+   */
+  public static CloseableIterable<Record> fromManifest(
+      Table table, ManifestFile manifest, Schema recordSchema) {
+    return CloseableIterable.transform(
+        ManifestFiles.open(manifest, table.io(), table.specs())
+            .select(BaseScan.scanColumns(manifest.content()))
+            .liveEntries(),
+        entry -> fromManifestEntry(entry, table, recordSchema));
+  }
+
+  /**
+   * Appends statistics from one Record to another.
+   *
+   * @param toRecord the Record to which statistics will be appended.
+   * @param fromRecord the Record from which statistics will be sourced.
+   */
+  public static void appendStatsFromRecord(Record toRecord, Record fromRecord) {
+    Preconditions.checkState(toRecord != null, "Record to update cannot be null");
+    Preconditions.checkState(fromRecord != null, "Record to update from cannot be null");
+
+    toRecord.set(
+        Column.SPEC_ID.ordinal(),
+        Math.max(
+            (int) toRecord.get(Column.SPEC_ID.ordinal()),
+            (int) fromRecord.get(Column.SPEC_ID.ordinal())));
+    checkAndIncrementLong(toRecord, fromRecord, Column.DATA_RECORD_COUNT);
+    checkAndIncrementInt(toRecord, fromRecord, Column.DATA_FILE_COUNT);
+    checkAndIncrementLong(toRecord, fromRecord, Column.TOTAL_DATA_FILE_SIZE_IN_BYTES);
+    checkAndIncrementLong(toRecord, fromRecord, Column.POSITION_DELETE_RECORD_COUNT);
+    checkAndIncrementInt(toRecord, fromRecord, Column.POSITION_DELETE_FILE_COUNT);
+    checkAndIncrementLong(toRecord, fromRecord, Column.EQUALITY_DELETE_RECORD_COUNT);
+    checkAndIncrementInt(toRecord, fromRecord, Column.EQUALITY_DELETE_FILE_COUNT);
+    checkAndIncrementLong(toRecord, fromRecord, Column.TOTAL_RECORD_COUNT);
+    if (fromRecord.get(Column.LAST_UPDATED_AT.ordinal()) != null) {
+      if (toRecord.get(Column.LAST_UPDATED_AT.ordinal()) == null
```

Comment:
> Hm, can we actually compute last_updated_at if at least one side has null? 

`fromRecord` side if it is null, then that snapshot might have expired and older than `toRecord` side. So, it won't be latest. Hence, no need to update. 

If `toRecord` side is null then `fromRecord` might be latest hence updating. 

So, when we go through all records for that partition, it will have the latest snapshot info.

Reply:
If we assume expired snapshots are older (which is mostly true unless someone expires a snapshot in the middle of the chain), then this logic seems to work.